units/provisioning_convert_host_to_legacy.go

```diff
@@ -0,0 +1,231 @@
+			"message": "could not populate required fields",
+			"host":    j.HostID,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+		j.AddError(err)
+		return
+	}
+
+	if j.host.NeedsReprovision != host.ReprovisionToLegacy || j.host.Status != evergreen.HostProvisioning {
+		return
+	}
+
+	// The host cannot be reprovisioned until the host's agent has
+	// stopped.
+	if !j.host.NeedsNewAgent {
+		grip.Error(message.WrapError(j.tryRequeue(ctx), message.Fields{
+			"message": "could not enqueue job to retry provisioning conversion when host's agent is still running",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+		return
+	}
+
+	defer func() {
+		if j.HasErrors() {
+			event.LogHostConvertingProvisioningError(j.host.Id, j.Error())
+
+			if err := j.tryRequeue(ctx); err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
+					"message": "could not enqueue job to retry provisioning conversion",
+					"host":    j.host.Id,
+					"distro":  j.host.Distro.Id,
+					"job":     j.ID(),
+				}))
+				j.AddError(err)
+			}
+			grip.Error(message.WrapError(j.Error(), message.Fields{
+				"message": "failed to convert host to legacy provisioning but will retry",
+				"host":    j.host.Id,
+				"distro":  j.host.Distro.Id,
+				"job":     j.ID(),
+			}))
+		}
+		grip.Error(message.WrapError(j.host.SetReprovisioningLocked(false), message.Fields{
+			"message": "could not clear host reprovisioning lock",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+	}()
+
+	// Lock the provisioning state to prevent other provisioning jobs from
+	// running.
+	if err := j.host.SetReprovisioningLockedAtomically(true); err != nil {
+		grip.Info(message.WrapError(err, message.Fields{
+			"message": "reprovisioning job currently in progress, returning from job",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+		return
+	}
+
+	if err := j.host.UpdateLastCommunicated(); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "could not update host communication time",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+	}
+
+	settings := j.env.Settings()
+	sshOpts, err := j.host.GetSSHOptions(settings)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "could not get SSH options",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+		j.AddError(err)
+		return
+	}
+	// This is a best-effort attempt to uninstall Jasper, but it will silently
+	// fail to uninstall Jasper if the Jasper binary path does not match its
+	// actual path on the remote host.
+	if logs, err := j.host.RunSSHCommandLiterally(ctx, fmt.Sprintf("[ -a \"%s\" ] && %s", j.host.JasperBinaryFilePath(settings.HostJasper), j.host.QuietUninstallJasperCommand(settings.HostJasper)), sshOpts); err != nil {
```

Comment:
I'd drop all of the grip.Errors here

Reply:
This would make it very difficult to diagnose issues since the SSH command is the most likely place of failure in this job and the only place where you can get the output logs.